activesupport/test/current_attributes_test.rb

```diff
@@ -0,0 +1,80 @@
+require "abstract_unit"
+require "active_support/current_attributes"
+
+Person = Struct.new(:name, :time_zone)
+
+class Current < ActiveSupport::CurrentAttributes
+  attribute :world, :account, :person
+
+  resets { Time.zone = "UTC" }
+
+  def account=(account)
+    attributes[:account] = account
+    self.person = "#{account}'s person"
+  end
+
+  def person=(person)
+    attributes[:person] = person
```

Comment:
Pushed a commit for this. Opted to not define the singleton methods within the module, since those should just be left be by users.